### DISCUSSION TITLE
Required workflows break default branch renaming

### DISCUSSION THREAD

**Discussion by bewuethr** [community/community#53252] [Enterprise] [answered]
### Select Topic Area

Bug

### Body

We just bumped into an interesting issue where having a required workflow broke default branch renaming.

Setup:

- The org-level default branch name is `main`
- We have a repo with default branch name `development`
- There is a required workflow that is applied to that repo, but because the default branch name doesn't match, it's not a required check
- The repo has branch protection rules for `development`

What we tried to do:

- Rename the default branch from `development` to `main`, via *Settings* > *General*

Expected outcome:

- Branch is renamed, including protection rule

Actual outcome:

- An HTTP 500 screen
- Branch isn't renamed
- A copy of the `development` branch protection rules called `main` is created

We tried a few things:

- Delete all branch protection rules; now, trying to rename the default branch didn't create any branch protection rules, but still failed with an HTTP 500
- Create a branch called `main` locally and push it, resulting in this error:

  ```none
  remote: error: GH013: Repository rule violations found for refs/heads/main.
  remote: Review all repository rules at http://github.com/ORG/REPO/rules?ref=refs%2Fheads%2Fmain
  remote:
  remote: - Organization mandated required workflow checks need to run successfully to be able merge to this branch. Please push changes to a new branch and create a pull request to run the checks.
  remote:
  To github.com:ORG/REPO.git
   ! [remote rejected]       main -> main (push declined due to repository rule violations)
  error: failed to push some refs to 'github.com:ORG/REPO.git'
  ```

At this point, we realized that it might be related to required workflows; disabling it for the repo let us rename the default branch.

**Comment by ghost** [comment]
Having a similar issue, except with creating a fresh repository within an organization with required workflows set to all repos. We cannot seem to create the main branch and when you try to create a new branch to merge into main, it just makes that branch the default automatically, and you cannot rename it to main. Even trying to add a README or .gitignore at creation time(by checking the boxes) fails.

Edit: Contacted Github Enterprise Support and they confirmed this is a bug and is being worked on.

**Reply by DebienRuben** [reply]
Any updates on this? Or do you have a link to a thread where they are writing updates of this issue. It's bothering us too with us being unable to create new repositories from templates as in the background it means the checks are failing or something.

**Reply by ghost** [reply]
@DebienRuben nope nothing from them yet. Our workaround is to set it to manual, but select all repos and then a daily slack reminder to make sure the list is kept up to date lol

**Reply by DebienRuben** [reply]
We're doing the same right now

**Comment by chticer** [comment]
I would also like to post an issue I am facing with required workflows from the organization level. Essentially, when I use GitHub Classroom to create a repository for an assignment with a required workflow, the repository is blank when it should create a pull request that is used for students and teachers. The entire discussion can be found here: https://github.com/community/Global-Campus-Teachers/discussions/222

**Comment by mtrunkat** [comment]
We bumped into the same problem. We have an organization-required workflow:
- targetted to all repositories
- run only on pull requests:

```
...

on:
  pull_request:
    types: ['opened', 'reopened', 'synchronize', 'labeled', 'unlabeled', 'edited', 'ready_for_review'] 

concurrency: 
  group: ${{ github.ref }}
  cancel-in-progress: true

...
```

and we are unable to create the **main** branch for a newly created repo. We have tried renaming, UI creation, change of a default branch, ... and nothing helped. The workaround to be able to continue is to create a branch with a different name (AKA old **master** branch) and switch the default branch to that one.

CC @mnmkng

**Comment by bewuethr** [comment, marked as answer]
It looks like this has been fixed, so I'll mark as resolved.

**Reply by mtrunkat** [reply]
Yes, the required workflows were deprecated by GitHub and removed and so the issue auto-resolved :).

**Reply by bewuethr** [reply]
And the replacement, a repository rule that requires a workflow to pass, works without the issues previously seen.
